File: src/main/java/org/example/expert/domain/todo/TodoQueryDslRepository.java
package org.example.expert.domain.todo;

import org.example.expert.controller.todo.dto.request.TodoListReqDto;
import org.example.expert.controller.todo.dto.request.TodoSearchReqDto;
import org.example.expert.domain.todo.Todo;
import org.springframework.data.domain.Page;
import org.springframework.data.domain.Pageable;

import java.util.Optional;

public interface TodoQueryDslRepository {
    Page<Todo> findTodosByFilter(TodoListReqDto todoListReqDto, Pageable pageable);

    Optional<Todo> findByIdWithUser(Long todoId);

    Page<TodoRespDto> searchTodosByFilter(TodoSearchReqDto todoSearchReqDto, Pageable pageable);
}

File: src/main/java/org/example/expert/domain/todo/TodoQueryDslRepository.kt
package org.example.expert.domain.todo

import org.example.expert.controller.todo.dto.request.TodoListReqDto
import org.example.expert.controller.todo.dto.request.TodoSearchReqDto
import org.example.expert.controller.todo.dto.response.TodoSearchRespDto
import org.example.expert.controller.todo.dto.response.TodoSearchRespDto.*
import org.springframework.data.domain.Page
import org.springframework.data.domain.Pageable

interface TodoQueryDslRepository {
    fun findTodosByFilter(todoListReqDto: TodoListReqDto, pageable: Pageable): Page<Todo>
    fun findByIdWithUser(todoId: Long): Todo?
    fun searchTodosByFilter(todoSearchReqDto: TodoSearchReqDto, pageable: Pageable): Page<TodoRespDto>
}